pkg/app/server/unregisteredappcache/cache.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2022 The PipeCD Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package unregisteredappcache
+
+import (
+	"bytes"
+	"context"
+	"encoding/gob"
+	"errors"
+	"fmt"
+
+	"go.uber.org/zap"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/pipe-cd/pipecd/pkg/cache"
+	"github.com/pipe-cd/pipecd/pkg/cache/rediscache"
+	"github.com/pipe-cd/pipecd/pkg/model"
+	"github.com/pipe-cd/pipecd/pkg/redis"
+)
+
+type Cache interface {
```

Comment:
Nice added 👍  But I think there is no reason to explicitly mention that it's a cache. How about renaming this interface as `Store` just like other `InsightStore` or `CommandStore` 👀

Reply:
At first I was going to use store, but I thought there was a difference between store, which is persistent, and cache, which is not. And since redis is not persistent, I thought it would be better to use cache.